### PR TITLE
basic scaffolding for cs prometheus docker image

### PR DESCRIPTION
**Pull request by nicktate**
@normanjoyner 

- Dynamically loads other prometheus servers and containership-prometheus-agents in the cluster for exporters. 

Still need to add in configuration tuning based on allocated docker memory for prometheus memory constraints.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+                let opts;
+
+                try {
+                    opts = JSON.parse(container.env_vars.CS_PROC_OPTS);
+                } catch (err) {
+                    return;
+                }
+
+                // Should I get the legiond scope?
```
Comment (normanjoyner) [inline lib/config.js]: yeah that would be preferred

File: lib/config.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        if (application && application.containers && application.containers.length) {
+            console.log('prometheus application');
+            console.log(JSON.stringify(application, null, ' '));
+
+            prometheusConfig.hosts = _.map(application.containers, container => {
+                const opts = JSON.parse(container.env_vars.CS_PROC_OPTS);
+
+                // Should I get the legiond scope?
```
Comment (normanjoyner) [inline lib/config.js]: yes - from local `CS_PROC_OPTS`

File: lib/config.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            console.log('no servers');
+            process.stderr.write(`${err.message}\n`);
+            return callback(null, prometheusConfig);
+        }
+
+        if (application && application.containers && application.containers.length) {
+            console.log('prometheus application');
+            console.log(JSON.stringify(application, null, ' '));
+
+            prometheusConfig.hosts = _.map(application.containers, container => {
```
Comment (normanjoyner) [inline lib/config.js]: will this be a problem if it grabs itself?

Comment (normanjoyner) [inline lib/config.js]: may be useful to filter out unloaded & loading containers here as well

File: lib/config.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        agent: getPrometheusAgentTargets
+    }, (error, results) => {
+        if (error) {
+            return generateConfigCallback(error);
+        }
+
+        let config =
+`
+global:
+    scrape_interval:     15s
```
Comment (normanjoyner) [inline lib/config.js]: can these be configurable via env var?

**Comment by normanjoyner** [comment]
can we also document all available environment variables, as well as their purpose in README

File: lib/config.js (hunk 3 above)
Comment (nicktate) [inline lib/config.js]: no, we want it to grab itself. prometheus server exposes a scraper so it can monitor itself

**Comment by nicktate** [comment]
@normanjoyner updated

**Pull request merged**